src/components/filterModal/statsList/StatsList.vue

```diff
@@ -90,6 +92,52 @@ export default {
                 }
             });
         },
+        hasFiltered(data) {
+            if (this.searchQuery) {
+                let array;
+                if (Array.isArray(data.data)) {
+                    array = data.data;
+                } else if (data.data.variants) {
+                    // INHERITANCE - custom handler
+                    array = data.data.variants.concat(data.data.family);
+                }
+                const result = array.filter(item => (Array.isArray(item) ? item[0].toLowerCase()
+                    .includes(this.searchQuery.toLowerCase()) : false));
+                return !!result.length;
+            }
+            return false;
+        },
+        hasSecondaryFiltered(data) {
+            return !!data.filter(item => (item.name ? item.name.toLowerCase()
+                .includes(this.searchQuery.toLowerCase()) : false)).length;
+        },
+        hasPrimaryFiltered(data) {
+            if (data.data && Array.isArray(data.data)) {
+                const filter = data.data.filter(item => (Array.isArray(item) ?
+                    this.hasFiltered(data) : this.hasFiltered(item)));
+                return filter.length ? true : this.hasSecondaryFiltered(data.data);
+            }
+            return false;
+        },
+        filteredData(stat) {
+            let result = stat.data;
+            if (stat.type !== 'zygosity' && Array.isArray(result)) {
+                result = this.filterData(result);
+            } else if (stat.type === 'zygosity') {
+                result = {
+                    ...result,
+                    variants: this.filterData(stat.data.variants),
+                    family: stat.data.family.filter(f =>
+                        f.toLowerCase().includes(this.searchQuery.toLowerCase())),
+                };
+            }
+            return result;
+        },
+        filterData(data) {
+            return data.filter(i => (Array.isArray(i) ? i[0].toLowerCase()
```

Comment:
Don's use 'i' name to perform the objects
Don't use a conditions chain - it's difficult to read.